Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -286,7 +286,7 @@ public Carousel()
                         carouselControl.SetValue(SelectedIndexProperty, index);
                     }
                 }
-                else if (SelectedItem == null && SelectedIndex >= 0)
+                else if (SelectedItem == null && SelectedIndex >= 0 && carouselControl.Items.Count > 0)
```

Comment:
Yes, that will be a better approach. I will push new commit with your suggestion.